lib/src/test/java/com/team2813/lib2813/util/InputValidationTest.java

```diff
@@ -1,38 +1,36 @@
 package com.team2813.lib2813.util;
 
-import static org.junit.Assert.assertEquals;
+import static com.google.common.truth.Truth.assertThat;
+import static com.team2813.lib2813.util.InvalidCanIdExceptionSubject.assertThat;
 import static org.junit.Assert.assertThrows;
 
-import com.team2813.lib2813.theories.Between;
-import org.junit.experimental.theories.Theories;
-import org.junit.experimental.theories.Theory;
-import org.junit.runner.RunWith;
+import org.junit.Test;
 
-@RunWith(Theories.class)
 public class InputValidationTest {
-  private static <T> RuntimeException outOfBounds(T val) {
-    return new RuntimeException("Value was out of bounds");
-  }
-
-  @Theory
-  public void invalidCanID(
-      @Between(first = -50, last = -1) @Between(first = 63, last = 100) int canID) {
-    InvalidCanIdException expected = new InvalidCanIdException(canID);
-    InvalidCanIdException actual =
-        assertThrows(InvalidCanIdException.class, () -> InputValidation.checkCanId(canID));
-    assertEquals("Messages were not identical", expected, actual);
-  }
-
-  @Theory
-  public void genericInvalidTest(@Between(first = 0, last = 4) int unused) {
-    Integer min = Integer.valueOf(0);
-    Integer actual = Integer.valueOf(unused);
-    Integer max = Integer.valueOf(4);
-    InputValidation.checkBounds(min, max, actual, InputValidationTest::outOfBounds);
-  }
+  // Tests for the `InputValidation.checkCanId(...)` method.
+  public static class checkCanIdTest {
+    @Test
+    public void invalidCanIdTest() {
+      // Can IDs can only valid in the range [0, 62].
+      int invalidCanIds[] = {-50, -1, 63, 100};
+      for (int invalidCanId : invalidCanIds) {
+        var exception =
+            assertThrows(
+                InvalidCanIdException.class, () -> InputValidation.checkCanId(invalidCanId));
+        assertThat(exception)
+            .hasCanId(invalidCanId)
+            .hasMessageThat()
+            .contains("is not a valid can id");
+      }
+    }
 
-  @Theory
-  public void validCanID(@Between(first = 0, last = 62) int canID) {
-    assertEquals(canID, InputValidation.checkCanId(canID));
+    @Test
+    public void validCanID() {
+      // Can IDs can only valid in the range [0, 62].
+      int validCanIds[] = {0, 1, 10, 62};
+      for (int validCanId : validCanIds) {
+        assertThat(InputValidation.checkCanId(validCanId)).isEqualTo(validCanId);
```

Comment:
Here it's a bit hard to see what code is actually being tested. Introducing a local and adding a blank line would make it easier to visually see what is being tested (and would be closer to the Arrange-Act-Assert pattern):

```suggestion
      int returnValue = InputValidation.checkCanId(validCanId);

      assertThat(returnValue).isEqualTo(validCanId);
```

Note that if this were to fail, the test failure message wouldn't indicate which input value was provided (it would in the Theories-based version). You could do something like this:

```suggestion
      int returnValue = InputValidation.checkCanId(validCanId);

      assertWithMessage("%d should be considered a valid ID")
          .that(returnValue).isEqualTo(validCanId);
```

I personally find the Theories-based version of this test to be extremely easy to read, but concede that having to create a custom `ParameterSupplier` increases the cost of using Theories in this class. I'm not convinced that we should forbid use of Theories.

Reply:
You're probably right. I'd suggest, let me get it my way this time just so I can gain a little more confidence with this code-base. I should not be allowed to do obviously wrong things in the code, but I hope it should be OK for me (or any other qualified contributor) to reshape it here and there to their liking ...